### ISSUE TITLE
Temporary directory invalid when running tests

### ISSUE THREAD

**Issue by JoeSzymanski** [calabash/calabash-ios#1244]
We've started testing with the new version of calabash and have hit an issue that we have not found a way to work around.

It looks like the new way of launching the apps for testing changes the environment such that the temporary directory available in the app is incorrect and cannot be written to.

When running the app directly, both `NSTemporaryDirectory()` and `-[NSFileManager URLForDirectory:inDomain:appropriateForURL:create:error:]` return a directory of  `/private/var/mobile/Containers/Data/Application/96317FAB-8342-4141-A392-22334178C3CD/tmp`

When running the tests, `NSTemporaryDirectory()` returns a directory of `/var/folders/k0/1063w_5j4_9dzgv4h179wny00000gn/T/` and `-[NSFileManager URLForDirectory:inDomain:appropriateForURL:create:error:]` returns a `nil` value. Running the Swift version of that code returns an error of:
> "You don’t have permission to save the file “Whitetail14A403.N61N71OS” in the folder “Whitetail14A403.N61N71OS”"

I have not been able to find any way to work around this issue, and I am reluctant to move the files into other locations, since those locations aren't really valid for temporary files.

Are there any ideas here on how we can move forward?

**Comment by JoeSzymanski** [comment]
I've managed to get a workaround for this by using the documents directory rather than the temp directory. My team is moving forward with this workaround, but we would still like to see if this issue can be resolved in any way, since we're no longer testing the same code path on the Calabash version of the app, and we do not want to store the temp files in the documents directory for the standard usage of the app.

**Comment by jmoody** [comment]
Ah.  I misunderstood the original problem.  I thought DeviceAgent was failing, but it sounds like somehow NSTemporaryDirectory is not returning the correct path when the application is under test.

Can you reproduce in the CalSmokeApp?

**Comment by JoeSzymanski** [comment]
Correct. It looks like NSTemporaryDirectory() is returning the temp directory for the DeviceAgent's sandbox, rather than the one for the app under test.

 I'll take a look at CalSmokeApp this afternoon and report back results.

**Comment by JoeSzymanski** [comment]
I'm not reproducing this issue in the CalSmokeApp so far. I'm going to look into how the app is launched versus how we are launching our app, since we may not be using the most up-to-date launching code.

I'll report back any findings here.

**Comment by JoeSzymanski** [comment]
@jmoody I took a look at the launching code in CalSmokeApp versus what we are using in our testing. The only difference I could find in the code was a few options to add to the launch call, but even setting those options gave an invalid temp directory when the tests ran.

I should point out that our current app testing is done in directories that don't have the Xcode project, but I'm not sure if that makes any difference here. Are there any other test cases you can think of that I should look into that could help track down why the temporary directory isn't working in our case?

**Comment by JoeSzymanski** [comment]
I believe that I've figured this out. When we started passing environment variables to the app to support launch configuration during testing, we were passing the entire ENV to the app. Apparently, this has a large number of settings that can break functionality of the app, including changing the temp dir.

Filtering the ENV based on the keys we want to send to the app looks like it fixes the problems we are seeing. Sorry about the false alarm.
